packages/ui/form/PhoneInput.tsx

```diff
@@ -18,11 +18,27 @@ export type PhoneInputProps = {
 };
 
 function BasePhoneInput({ name, className = "", onChange, ...rest }: PhoneInputProps) {
+  useEffect(() => {
+    if (!rest.value) {
+      return;
+    }
+    const formattedValue = rest.value.trim().replace(/^\+?/, "+");
+    onChange(formattedValue);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  let countryCode;
+  if (rest.value) {
+    try {
+      countryCode = parsePhoneNumber(rest.value)?.country || "";
+    } catch {}
+  }
+
```

Comment:
I don't think we need this

Reply:
@Udit-takkar I will remove this if you can answer the below query of mine. Suppose I am in US country right now, But I pre-filled phone number that is outside US ( for example : `India`) In that case even if according to phone number it was supposed to show India country flag on `Dropdown` it will show `US` flag. So is that ok for you? If not then we don't need to remove this logic.

See the attached video what I am trying to say. See I filled the Indian number but it selected US. Then I tried to edit it, it changed to India Flag. I think this will be wrong behaviour

https://github.com/calcom/cal.com/assets/37592944/a49aee06-498d-49df-92f0-b44b1b2339ee

Reply:
No it should show the correct flag.

Reply:
So, @Udit-takkar we are okay to keep this logic or you want me to remove it?

Reply:
you are not passing countryCode anywhere and i tried removing this logic and it still works fine with the correct flag.

<img width="549" alt="Screenshot 2023-07-01 at 4 36 02 PM" src="https://github.com/calcom/cal.com/assets/53316345/6522f91d-9e79-482c-b741-e3ceb579809e">

Reply:
No @Udit-takkar that logic was actually checking if my pre filled phone number is correct than only replace it with undefined on country input, other wise keep the default country flag in the input. See now we have removed that logic this will happen for wrong phone number, User will see no flag on the dropdown. Is that ok?

https://github.com/calcom/cal.com/assets/37592944/1ff5064e-93e4-44ff-8af1-9003a1b4b518

Reply:
yeah that's fine for wrong number

Reply:
Video:  When we don't remove that logic. User will still see the default country flag even with wrong phone number.

https://github.com/calcom/cal.com/assets/37592944/172a59b2-6966-48af-be46-24388c59c477

Reply:
> yeah that's fine for wrong number

Aah ok @Udit-takkar, then that's fine and sorry if I sounded harsh I am not that good with english 😄

Reply:
> Aah ok @Udit-takkar, then that's fine and sorry if I sounded harsh I am not that good with english 😄

Not at all. Thanks for contribution